### PR TITLE
Support server parameter in modular

### LINKED ISSUE #2372: [modular] fail to generate anomalyDetector code with isModularLibrary: true
for anomalyDetector the generation error is:
image
code details :
```
detectMultivariateBatchAnomaly: (
    modelId: string,
    options?: MultivariateMultivariateBatchDetectionOptions,
    options?: MultivariateDetectMultivariateBatchAnomalyOptions,
  ) => Promise<MultivariateMultivariateDetectionResult>;
  detectMultivariateLastAnomaly: (
    modelId: string,
    options?: MultivariateMultivariateLastDetectionOptions,
    options?: MultivariateDetectMultivariateLastAnomalyOptions,
  ) => Promise<MultivariateMultivariateLastDetectionResult>;
```

### LINKED ISSUE #2376: [modular] fail to generate parametrizedHost code with isModularLibrary: true
error:
```
src/api/ParametrizedHostContext.ts(20,52): error TS2554: Expected 1-2 arguments, but got 5.
```
image

code:
```
export default function createClient(
  credentials: TokenCredential,
  options: ParametrizedHostContextOptions = {},
): ParametrizedHostContext {
...
```

### PR DESCRIPTION
**Pull request by qiaozha**
Support server parameter in modular and enable modular in parameterized host and anomaly detector. Also, I found we didn't normalize the filename for modular client and classical client, so updated in this pr. 

fix #2372
fix #2376
fix https://github.com/Azure/autorest.typescript/issues/2368

### CODE REVIEW

**Comment by qiaozha** [comment]
pending https://github.com/Azure/autorest.typescript/pull/2387 to enable modular for anomalydetector as of the duplicate options name.

File: packages/typespec-test/test/anomalyDetector/generated/typespec-ts/src/api/AnomalyDetectorContext.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ * service user can provide labels for each detection result, these labels will be
+ * used for retuning or regenerating detection models. Inconsistency detection is
+ * a kind of group based detection, this detection will find inconsistency ones in
+ * a set of time series. By using anomaly detector service, business customers can
+ * discover incidents and establish a logic flow for root cause analysis.
+ */
+export function createAnomalyDetector(
+  endpointParam: string,
+  credential: KeyCredential,
+  options: AnomalyDetectorClientOptions = {},
```
Comment (MaryGao) [inline packages/typespec-test/test/anomalyDetector/generated/typespec-ts/src/api/AnomalyDetectorContext.ts:34]: why there is no apiVersion setting in classical client?

File: packages/typespec-test/test/anomalyDetector/generated/typespec-ts/src/api/AnomalyDetectorContext.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+): AnomalyDetectorContext {
```
Comment (MaryGao) [inline packages/typespec-test/test/anomalyDetector/generated/typespec-ts/src/api/AnomalyDetectorContext.ts:35]: We have a cadl-ranch case for this, could you help check this is covered by modular?

https://github.com/Azure/cadl-ranch/blob/main/packages/cadl-ranch-specs/http/server/path/multiple/main.tsp#L21

File: packages/typespec-test/test/anomalyDetector/generated/typespec-ts/src/api/AnomalyDetectorContext.ts (hunk 1 above)
Comment (qiaozha) [reply packages/typespec-test/test/anomalyDetector/generated/typespec-ts/src/api/AnomalyDetectorContext.ts:37]: added.

**Review by joheredi — APPROVED** [review]
We should try to start thinking away from smoke-tests. Ideally for these changes we would introduce unit tests that validates the behavior of the code. This would make the tests more reliable and faster to run.

I'm preparing a PR restructuring a little bit the test suite and adding a few unit-tests that can serve as guide for future tests. In the meantime, take this comment as a heads up to what's coming in the next days.

**Pull request merged**